src/views/scans/viewScansList.tsx

```diff
@@ -286,80 +253,73 @@ const ScansListView: React.FunctionComponent = () => {
         </ConditionalTableBody>
       </Table>
       <Pagination variant="bottom" widgetId="server-paginated-example-pagination" />
-      {!!scanSelectedForSources && (
-        <Modal
-          variant={ModalVariant.small}
-          title={t('view.label', { context: 'sources' })}
-          isOpen={!!scanSelectedForSources}
-          onClose={() => setScanSelectedForSources(undefined)}
-          actions={[
-            <Button key="cancel" variant="secondary" onClick={() => setScanSelectedForSources(undefined)}>
-              Close
-            </Button>
-          ]}
-        >
-          <List isPlain isBordered>
-            {scanSelectedForSources.sources.map(s => (
-              <ListItem key={s.id}>{s.name}</ListItem>
-            ))}
-          </List>
-        </Modal>
-      )}
-      {!!scanSelected && (
+      <Modal
+        variant={ModalVariant.small}
+        title={t('view.label', { context: 'sources' })}
+        isOpen={scanSelectedForSources !== undefined}
+        onClose={() => setScanSelectedForSources(undefined)}
+        actions={[
+          <Button
+            key="confirm"
+            variant="danger"
+            onClick={() => {
+              setScanSelectedForSources(undefined);
+            }}
+          >
+            {t('table.label', { context: 'close' })}
+          </Button>,
+          <Button key="cancel" variant="link" onClick={() => setScanSelectedForSources(undefined)}>
+            {t('form-dialog.label', { context: 'cancel' })}
+          </Button>
+        ]}
+      >
+        <List isPlain isBordered>
+          {scanSelectedForSources?.sources.map(s => <ListItem key={s.id}>{s.name}</ListItem>)}
+        </List>
+      </Modal>
+      {scanSelected && (
         <ScansModal
           scan={scanSelected}
           scanJobs={scanJobs}
-          onDownload={reportId => {
-            downloadReport(reportId)
-              .then(res => {
-                addAlert({
-                  title: `Report "${reportId}" downloaded`,
-                  variant: 'success',
-                  id: getUniqueId()
-                });
-                helpers.downloadData(
-                  res.data,
-                  `report_id_${reportId}_${new Date()
-                    .toISOString()
-                    .replace('T', '_')
-                    .replace(/[^\d_]/g, '')}.tar.gz`,
-                  'application/gzip'
-                );
-              })
-              .catch(err => {
-                console.error(err);
-                addAlert({
-                  title: `Report "${reportId}" failed to download. ${JSON.stringify(err.response.data)}`,
-                  variant: 'danger',
-                  id: getUniqueId()
-                });
-              });
+          onDownload={(reportId: number) => {
+            downloadReport(reportId);
```

Comment:
optional:
you could simplify this to `onDownload={downloadReport}`... having to "redeclare" the param type is a little typescript